src/components/routes/CalendarItemDisplay.vue

```diff
@@ -0,0 +1,241 @@
+import { useRoutes } from 'src/composables/useRoutes';
+
+// types
+import type { Timestamp } from '@quasar/quasar-ui-qcalendar';
+import type {
+  RouteCalendarDay,
+  RouteItem,
+  TransportDirection,
+} from '../types/Route';
+
+export default defineComponent({
+  name: 'CalendarItemDisplay',
+  props: {
+    day: {
+      type: Object as () => RouteCalendarDay | null,
+      default: null,
+    },
+    direction: {
+      type: String as () => TransportDirection,
+      required: true,
+    },
+    timestamp: {
+      type: Object as () => Timestamp,
+      required: true,
+    },
+    active: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  emits: ['item-click'],
+  setup(props, { emit }) {
+    function onClick(): void {
+      emit('item-click', {
+        timestamp: props.timestamp,
+        direction: props.direction,
+      });
+    }
+
+    const route = computed((): RouteItem | null => {
+      if (!props.day) return null;
+      return props.day[props.direction] ? props.day[props.direction] : null;
+    });
+
+    const { getRouteIcon } = useRoutes();
+
+    return {
+      route,
+      getRouteIcon,
+      onClick,
+    };
+  },
+});
+</script>
+
+<template>
+  <div class="q-my-sm" data-cy="calendar-item-display">
+    <q-item
+      :active="active"
+      dense
+      clickable
+      v-ripple
+      class="relative-position flex justify-center items-center text-center gap-8"
+      @click.prevent="onClick()"
+      data-cy="calendar-item-display-item"
+    >
+      <!-- Background: To work -->
+      <template v-if="direction === 'toWork'">
+        <!-- Icon: To work - active -->
+        <q-icon
+          v-if="active"
+          color="primary"
+          class="full-width full-height absolute-full"
+          data-cy="calendar-item-icon-towork-active"
+        >
+          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 135 38">
+            <path
+              fill-rule="evenodd"
+              d="M8.857 0a8 8 0 0 0-8 8v22a8 8 0 0 0 8 8h99.2l26.8-19-26.8-19h-99.2Z"
+              clip-rule="evenodd"
+            />
+          </svg>
+        </q-icon>
+        <!-- Icon: To work - logged -->
+        <q-icon
+          v-else-if="route && route.id"
+          color="secondary"
+          class="full-width full-height absolute-full"
+          style="opacity: 0.4"
+          data-cy="calendar-item-icon-towork-logged"
+        >
+          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 135 38">
+            <path
+              fill-rule="evenodd"
+              d="M8.857 0a8 8 0 0 0-8 8v22a8 8 0 0 0 8 8h99.2l26.8-19-26.8-19h-99.2Z"
+              clip-rule="evenodd"
+            />
+          </svg>
+        </q-icon>
+        <!-- Icon: To work - empty -->
+        <q-icon
+          v-else
+          color="primary"
+          class="full-width full-height absolute-full"
+          style="opacity: 0.5"
+          data-cy="calendar-item-icon-towork-empty"
+        >
+          <svg
```

Comment:
Should be standalone SVG image.

Reply:
Refactored using [public svg with svg:use](https://quasar.dev/vue-components/icon#svg-use-way)
Will be used for new custom SVGs.